File: Exercises/101-109 - Small Problems/easy_7/capitalize_words.rb
# Write a method that takes a single String argument and returns a new string that contains
# the original value of the argument, but the first letter of every word is now capitalized.

# You may assume that words are any sequence of non-blank characters, and that only the
# first character of each word must be considered.

# Examples

# ```
# word_cap('four score and seven') == 'Four Score And Seven'
# word_cap('the javaScript language') == 'The Javascript Language'
# word_cap('this is a "quoted" word') == 'This Is A "quoted" Word'
# ```

def word_cap(words)
  words.split.map { |word| word.capitalize }.join(' ')
end

puts word_cap('four score and seven') == 'Four Score And Seven'
puts word_cap('the javaScript language') == 'The Javascript Language'
puts word_cap('this is a "quoted" word') == 'This Is A "quoted" Word'

File: Exercises/210 - Small Problems/string-text-processing/capitalize-words.js
/*
Write a function that takes a single string argument and returns a new string that contains the original value of the argument with the first character of every word capitalized and all succeeding characters in lowercase.

You may assume that words are any sequence of non-blank characters.

Examples

wordCap('four score and seven')    // 'Four Score And Seven'
wordCap('the javaScript language') // 'The Javascript Language'
wordCap('this is a "quoted" word') // 'This Is A "quoted" Word'
*/

function wordCap(str) {
  return str.split(' ').map(function(word) {
    return word.slice(0, 1).toUpperCase() + word.slice(1).toLowerCase();
  }).join(' ');
}

console.log(wordCap('four score and seven'));
console.log(wordCap('the javaScript language'));
console.log(wordCap('this is a "quoted" word'));